shop/src/components/payment-widget.js

```diff
@@ -0,0 +1,168 @@
+/* global AlgoSigner */
+import './signer.css';
+
+import { Button, CircularProgress, Typography } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useCallback, useState } from 'react';
+
+import { CHAIN_NAME } from '../algosigner.config';
+import { waitForConfirmation } from '../utils/algod';
+
+const LAST_ROUND = 'last-round';
+const CONFIRMED_ROUND = 'confirmed-round';
+
+/**
+ * Get default account address from user's wallet. Returns undefined otherwise.
+ * NOTE: Currently this returns the first address from list of account addresses
+ *
+ * TODO: Update this function to return "current active" account in wallet after
+ * https://github.com/PureStake/algosigner/issues/252 is resolved.
+ * Task - https://www.pivotaltracker.com/story/show/178760753
+ */
+async function getDefaultAccountAddr() {
+  const walletAccounts =
+    (await AlgoSigner.accounts({
+      ledger: CHAIN_NAME,
+    })) ?? [];
+
+  if (walletAccounts.length) {
+    return walletAccounts[0].address;
+  }
+  return undefined;
+}
+
+/**
+ * Executes transaction to receive payment of ticket
+ * @param fromAddress address of currently active account in wallet
+ * @param {*} toAddress address to recieve payment (currently set to master-account)
```

Comment:
what is `{*}` ?

Reply:
used to represent  type information. removing